nexus/src/authn/external/spoof.rs

```diff
@@ -13,21 +13,59 @@ use crate::authn::Actor;
 use anyhow::anyhow;
 use anyhow::Context;
 use async_trait::async_trait;
+use headers::authorization::{Authorization, Bearer};
+use headers::Header;
+use headers::HeaderMapExt;
 use lazy_static::lazy_static;
 use uuid::Uuid;
 
-/// Header used for "spoof" authentication
-pub const HTTP_HEADER_OXIDE_AUTHN_SPOOF: &str = "oxide-authn-spoof";
-/// Magic header value to produce a "no such actor" error
-pub const SPOOF_RESERVED_BAD_ACTOR: &str = "Jack Donaghy";
-/// Magic header value to produce a "bad credentials" error
-pub const SPOOF_RESERVED_BAD_CREDS: &str =
-    "this fake I.D., it is truly excellent";
+// This scheme is intended for demos, development, and testing until we have a
+// more automatable identity provider that can be used for those purposes.
+//
+// For ease of integration into existing clients, we use RFC 6750 bearer tokens.
+// This mechanism in turn uses HTTP's "Authorization" header.  In practice, it
+// looks like this:
+//
+//     Authorization: Bearer oxide-spoof-001de000-05e4-4000-8000-000000060001
+//     ^^^^^^^^^^^^^  ^^^^^^ ^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+//     |              |      |           |
+//     |              |      |           +--- specifies the actor id
+//     |              |      +--------------- specifies this "spoof" mechanism
+//     |              +---------------------- specifies RFC 6750 bearer tokens
+//     +------------------------------------- standard HTTP authentication hdr
+//
+// (That's not a typo -- the "authorization" header is generally used to specify
+// _authentication_ information.  Similarly, the "Unauthorized" HTTP response
+// code usually describes an _authentication_ error.)
+//
+// **This mechanism trusts (without verification) that the client is whoever
+// they say they are.**
```

Comment:
Pedantic thought, but this doesn't really make it different from the usual bearer token, it just happens that the token is the user UUID and therefore long-lived and publicly exposed.

Reply:
Fair point.  I updated the comment.